pkgs/by-name/go/goto/package.nix

```
@@ -0,0 +1,65 @@
+{
+  coreutils,
+  fetchFromGitHub,
+  gawk,
+  goto,
+  lib,
+  nix-update-script,
+  runCommand,
+  stdenvNoCC,
+}:
+
+let
+  owner = "iridakos";
+  rev = "b7fda54e0817b9cb47e22a78bd00b4571011cf58";
+
+in
+stdenvNoCC.mkDerivation {
+  pname = "goto";
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    inherit owner rev;
+    repo = "goto";
+    hash = "sha256-dUxim8LLb+J9cI7HySkmC2DIWbWAKSsH/cTVXmt8zRo=";
+  };
+
+  strictDeps = true;
+
+  buildInputs = [
+    coreutils
+    gawk
+  ];
+
+  postInstall = ''
+    install -Dm644 goto.sh -t $out/share/
+  '';
+
+  passthru.tests.basic-usage = runCommand "goto-basic-usage" { } ''
+    # Mock `complete` since the builder `pkgs.bash` is not interactive.
+    complete() { return; }
+
+    # Mock `$XDG_CONFIG_HOME` since `$HOME=/homeless-shelter` does not exist.
+    export XDG_CONFIG_HOME="$(mktemp -d)"
+
+    source ${goto}/share/goto.sh
+
+    goto --register pwd .
+    cd /
+    goto pwd
+    goto --unregister pwd
+    goto --list
+
+    touch $out
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = {
+    description = "Alias and navigate to directories with tab completion";
+    homepage = "https://github.com/${owner}/goto";
+    changelog = "https://github.com/${owner}/goto/blob/${rev}/CHANGELOG.md";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.bmrips ];
+  };
+}
```

```suggestion
})
```
(`finalAttrs` end parenthesis)